layers/+distributions/spacemacs-base/local/centered-buffer-mode/centered-buffer-mode.el

```diff
@@ -0,0 +1,128 @@
+;;; centered-buffer-mode.el --- Minor mode for centering buffers.
+
+;; Copyright (C) 2012-2016 Sylvain Benner & Contributors
+;;
+;; Author: Sylvain Benner <sylvain.benner@gmail.com>
+;; Keywords: centering
+;; Created: 1 July 2016
+;; Version: 1.00
+;; Package-Requires: ((emacs "24.4"))
+;; URL: https://github.com/syl20bnr/spacemacs
+
+;; This file is not part of GNU Emacs.
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Code:
+(require 'face-remap)
+
+(defvar-local spacemacs--centered-buffer-mode-origin-buffer nil)
+(defvar-local spacemacs--centered-buffer-mode-indirect-buffer nil)
+(defvar-local spacemacs--centered-buffer-mode-text-pixel-size nil)
+(defvar spacemacs--centered-buffer-mode-min-fringe-width 50)
+
+(define-minor-mode spacemacs-centered-buffer-mode
+  "Minor mode to center the current buffer.
+
+Hide line number and empty lines indicator(~) if
+called interactively."
```

Comment:
>Ok thank you for mentioning it. If you plan to fix it soon maybe we can wait for the fix before merging this PR.  
@syl20bnr  Fixed most of it in #6507 except fringe color corruption.  It will take some time. I'll probably ask Emacs devs what's going on.